python/src/schema.rs

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use arrow::pyarrow::PyArrowType;
+use arrow_schema::Schema as ArrowSchema;
+use lance::datatypes::{Field, Schema};
+use lance_file::datatypes::Fields;
+use lance_file::format::pb;
+use prost::Message;
+use pyo3::{
+    basic::CompareOp,
+    exceptions::{PyNotImplementedError, PyValueError},
+    prelude::*,
+    types::PyTuple,
+};
+
+/// A Lance Schema.
+///
+/// This is valid for a particular dataset. It contains the field ids for each
+/// column in the dataset.
```

Comment:
This doesn't have to be in this PR.  However, if we are introducing the concept of `LanceSchema` it would probably be good to provide some more motivation here.

Something like:

In order to support schema evolution each field in a lance dataset has an "id", this allows...